pkg/controller/reconciler/annotations.go

```diff
@@ -52,6 +52,11 @@ type deploymentAnnotations struct {
 	Status *deploymentStatus
 }
 
+// Case 1: Enabled => Disabled
+// Case 2: Disabled => Enabled
+// Case 3: Enabled => Enabled
+// Case 4: Disabled => Disabled
```

Comment:
I think this comment needs more context to be useful. Imagine if you came back to this in 6 months, how would you know what it's saying?